pkg/pool-utils/contracts/controllers/BasePoolController.sol

```diff
@@ -0,0 +1,99 @@
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+pragma solidity ^0.7.0;
+
+import "@balancer-labs/v2-solidity-utils/contracts/openzeppelin/Ownable.sol";
+
+import "../BasePoolAuthorization.sol";
+import "../interfaces/IBasePoolController.sol";
+import "../interfaces/IControlledPool.sol";
+
+/**
+ * @dev Pool controller that serves as the "owner" of a Balancer pool, and is in turn owned by
+ * an account empowered to make calls on this contract that are forwarded to the underlyling pool.
+ *
+ * While Balancer pool owners are immutable, ownership of this pool controller is transferrable.
+ * The deployer will be the initial owner
+ */
+contract BasePoolController is IBasePoolController, Ownable {
+    address public pool;
+
+    // Optional metadata associated with this controller (or the pool bound to it)
+    bytes private _metadata;
+
+    event MetadataUpdated(bytes metadata);
+
+    modifier withBoundPool {
+        _ensurePoolIsBound();
+        _;
+    }
+
+    /**
+     * @dev The pool needs this controller's address (as its owner), and this controller also needs the
+     * pool's address to delegate the calls.
+     */
+    function bindPool(address poolAddress) external virtual override {
+        _require(
+            pool == address(0) && BasePoolAuthorization(poolAddress).getOwner() == address(this),
+            Errors.INVALID_INITIALIZATION
+        );
+
+        pool = poolAddress;
+    }
+
+    /**
+     * @dev Pass a call to BasePool's setSwapFeePercentage through to the underlying pool
+     */
+    function setSwapFeePercentage(uint256 swapFeePercentage) external virtual override onlyOwner withBoundPool {
+        IControlledPool(pool).setSwapFeePercentage(swapFeePercentage);
+    }
+
+    /**
+     * @dev Pass a call to BasePool's setAssetManagerPoolConfig through to the underlying pool
+     */
+    function setAssetManagerPoolConfig(IERC20 token, bytes memory poolConfig)
+        external
+        virtual
+        override
+        onlyOwner
+        withBoundPool
+    {
+        IControlledPool(pool).setAssetManagerPoolConfig(token, poolConfig);
+    }
+
+    /**
+     * @dev Getter for the optional metadata
+     */
+    function getMetadata() public view returns (bytes memory) {
+        return _metadata;
+    }
+
+    /**
+     * @dev Setter for the admin to set/update the metadata
+     */
+    function updateMetadata(bytes memory metadata) external onlyOwner {
+        _updateMetadata(metadata);
+    }
+
+    function _updateMetadata(bytes memory metadata) internal virtual {
+        _metadata = metadata;
+
+        emit MetadataUpdated(metadata);
+    }
```

Comment:
This came from reviewing other controllers (specifically PrimeDAO's LBP). In V1 we stored offchain metadata associated with smart pools ("Learn more" link, etc.) PrimeDAO stores an IPFS hash of the metadata used when creating the LBP in their controller, so I thought it might be useful to enable this here (vs in the underlying pools, or off-chain).

I didn't want to force people to pass in 0x if they weren't using it, so I made it default to unused (no constructor), and provided a means to add metadata if desired.

Up for discussion - if the consensus is we don't need this (or don't need it at the base level), I can move it down to the ManagedPool controller, or just take it out

Reply:
I find it odd that this would be stored in the contract, but if it's been used let's keep it I guess?